WarehouseManagement.Core/Services/MarkerService.cs

```diff
@@ -0,0 +1,213 @@
+
+    public async Task<IEnumerable<MarkerDto>> GetAllAsync()
+    {
+        return await repository
+            .AllReadOnly<Marker>()
+            .Select(m => new MarkerDto
+            {
+                Name = m.Name,
+                Deliveries = m
+                    .DeliveriesMarkers.Select(dm => new MarkerDeliveryDto
+                    {
+                        DeliveryId = dm.DeliveryId,
+                        DeliverySystemNumber = dm.Delivery.SystemNumber
+                    })
+                    .ToList(),
+                Vendors = m
+                    .VendorsMarkers.Select(vm => new MarkerVendorDto
+                    {
+                        VendorId = vm.VendorId,
+                        VendorName = vm.Vendor.Name,
+                        VendorSystemNumber = vm.Vendor.SystemNumber
+                    })
+                    .ToList(),
+                Zones = m
+                    .ZonesMarkers.Select(zm => new MarkerZoneDto
+                    {
+                        ZoneId = zm.ZoneId,
+                        ZoneName = zm.Zone.Name
+                    })
+                    .ToList()
+            })
+            .ToListAsync();
+    }
+
+    public async Task<MarkerDto?> GetByIdAsync(int id)
+    {
+        return await repository
+            .AllReadOnly<Marker>()
+            .Where(m => m.Id == id)
+            .Select(m => new MarkerDto
+            {
+                Name = m.Name,
+                Deliveries = m
+                    .DeliveriesMarkers.Select(dm => new MarkerDeliveryDto
+                    {
+                        DeliveryId = dm.DeliveryId,
+                        DeliverySystemNumber = dm.Delivery.SystemNumber
+                    })
+                    .ToList(),
+                Vendors = m
+                    .VendorsMarkers.Select(vm => new MarkerVendorDto
+                    {
+                        VendorId = vm.VendorId,
+                        VendorName = vm.Vendor.Name,
+                        VendorSystemNumber = vm.Vendor.SystemNumber
+                    })
+                    .ToList(),
+                Zones = m
+                    .ZonesMarkers.Select(zm => new MarkerZoneDto
+                    {
+                        ZoneId = zm.ZoneId,
+                        ZoneName = zm.Zone.Name
+                    })
+                    .ToList()
+            })
+            .FirstOrDefaultAsync();
+    }
+
+    public async Task RestoreAsync(int id, string userId)
+    {
+        var marker = await repository.GetByIdWithDeletedAsync<Marker>(id);
+
+        if (marker == null)
+        {
+            throw new KeyNotFoundException($"Marker with ID {id} not found.");
+        }
+
+        marker.LastModifiedByUserId = userId; //TODO: Should the user that undeleted be assigned to LastModified user ?
+        repository.UnDelete(marker);
+
+        await repository.SaveChangesWithLogAsync(); //TODO: Should we log the Undeletion process or not ?
```

Comment:
dont think we should also log this so maybe just use the normal SaveChanges and remove the TODO?

Reply:
Okay. Replaced it with the normal SaveChangesAsync method.